### PR TITLE
Respond with error if file/directory is not found

### LINKED ISSUE #3: Should not fallback to index.html when the non-existent path has an extension
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/electron-serve/issues/3)

And the extension is not `.html`.

Loading `index.html` on these requests, makes it really hard to debug problems.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)
### Submitted pull Requests
- [#13 Respond with error if file/directory is not found](https://issuehunt.io/r/sindresorhus/electron-serve/pull/13)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/electron-serve/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by hadeeb**
closes #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,12 +33,15 @@ module.exports = options => {
...
 
-		callback({
-			path: (await getPath(filePath)) || indexPath
-		});
+		const resolvedPath = await getPath(filePath);
+
+		if (resolvedPath) {
+			callback({path: resolvedPath});
+		} else {
+			callback({error: -6});
```
Comment (sindresorhus) [inline index.js]: Where did you get this number from? It should be moved into a variable with a descriptive name.

**Comment by sindresorhus** [comment]
This doesn't exactly solve what the #3 is describing.

File: index.js (hunk 1 above)
Comment (hadeeb) [reply index.js]: It's documented here 
[RegisterFileProtocol](https://github.com/electron/electron/blob/master/docs/api/protocol.md#protocolregisterfileprotocolscheme-handler-completion)
[net error list](https://cs.chromium.org/chromium/src/net/base/net_error_list.h)

Found a similar implementation here [electron-routes](https://github.com/MarshallOfSound/electron-routes/blob/4240bffd875be38dc1538de7db27d0bd84e18f2f/src/Router.js#L84)

**Comment by sindresorhus** [comment]
Is this ready to be reviewed again?

**Comment by hadeeb** [comment]
Yes

File: test/test.js [hunk 2]
```diff
@@ -30,3 +30,26 @@ test('allows special characters in file paths', async t => {
...
+
+test('throws error if unresolved path has an extension other than .html', async t => {
+	t.context.spectron = new Application({
+		path: electron,
+		args: ['fixture-404-error.js']
+	});
+	await t.context.spectron.start();
+	const {client} = t.context.spectron;
+	await client.waitUntilWindowLoaded();
+	await t.throwsAsync(client.waitUntilTextExists('h1', '🦄', 5000));
```
Comment (sindresorhus) [inline test/test.js:54]: You used the assertion incorrectly, and when fixed (https://github.com/sindresorhus/electron-serve/pull/13/commits/037261161e8e2a07fe0cfa3214bde415d25bede0), it no longer passes.

Comment (hadeeb) [reply test/test.js:54]: Upgrading `ava` to v2 fixed the issue and the test cases are passing.

I don't know if it is out of scope of this PR.

Comment (sindresorhus) [reply test/test.js:54]: That's fine :)

**Pull request merged**